# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/CreateElementTransaction.h
# Commit: 876b1031a6ae
# Full Hash: 876b1031a6ae41a87ed146976f279b96bde9c7f6
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2020-04-03 21:40:10
# Regressor Bug: 1619914
# File Overlap Count: 1
# Description:
#   Bug 1619914 - part 1: Make each transaction class grab their members with local variable before touching the DOM tree r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D69302
# ==============================================================================

diff -r 7f069649d6cf -r 876b1031a6ae editor/libeditor/CreateElementTransaction.h
--- a/editor/libeditor/CreateElementTransaction.h	Fri Apr 03 06:03:25 2020 +0300
+++ b/editor/libeditor/CreateElementTransaction.h	Fri Apr 03 01:38:22 2020 +0000
@@ -8,13 +8,12 @@
 
 #include "mozilla/EditorDOMPoint.h"
 #include "mozilla/EditTransactionBase.h"
-#include "nsCOMPtr.h"
+#include "mozilla/RefPtr.h"
+#include "mozilla/dom/Element.h"
 #include "nsCycleCollectionParticipant.h"
 #include "nsISupportsImpl.h"
 
 class nsAtom;
-class nsIContent;
-class nsINode;
 
 /**
  * A transaction that creates a new node in the content tree.
@@ -22,9 +21,6 @@
 namespace mozilla {
 
 class EditorBase;
-namespace dom {
-class Element;
-}  // namespace dom
 
 class CreateElementTransaction final : public EditTransactionBase {
  protected:
@@ -57,7 +53,7 @@
 
   NS_IMETHOD RedoTransaction() override;
 
-  already_AddRefed<dom::Element> GetNewNode();
+  dom::Element* GetNewElement() const { return mNewElement; }
 
  protected:
   virtual ~CreateElementTransaction() = default;
@@ -77,7 +73,7 @@
   EditorDOMPoint mPointToInsert;
 
   // The new node to insert.
-  nsCOMPtr<dom::Element> mNewNode;
+  RefPtr<dom::Element> mNewElement;
 };
 
 }  // namespace mozilla